### PR TITLE
Bicaws7-2436: Add user notes to reallocation page

### PR DESCRIPTION
**Pull request by ian-antking**
Before:
Screenshot 2024-01-22 at 11 08 09

After:
Screenshot 2024-01-22 at 11 04 43
Screenshot 2024-01-22 at 11 04 59
Screenshot 2024-01-22 at 11 05 13

### CODE REVIEW

File: src/components/ReallocationNotesForm.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+const ReallocationNotesForm = ({ courtCase, backLink }: Props) => {
+  const [noteRemainingLength, setNoteRemainingLength] = useState(MAX_NOTE_LENGTH)
+  const classes = useCustomStyles()
+  const currentForce = forces.find((force) => force.code === courtCase.orgForPoliceFilter?.substring(0, 2))
+  const forcesForReallocation = getForcesForReallocation(currentForce?.code)
+  const handleOnNoteChange: FormEventHandler<HTMLTextAreaElement> = (event) => {
+    setNoteRemainingLength(MAX_NOTE_LENGTH - event.currentTarget.value.length)
+  }
+
+  const csrfToken = useContext(CsrfTokenContext)
```
Comment (richrace) [inline src/components/ReallocationNotesForm.tsx:26]: See `useCsrfToken` instead

File: src/components/ReallocationNotesForm.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import Form from "./Form"
+import { useCustomStyles } from "../../styles/customStyles"
+import { FormEventHandler, useContext, useState } from "react"
+import { forces } from "@moj-bichard7-developers/bichard7-next-data"
+import getForcesForReallocation from "services/getForcesForReallocation"
+import { DisplayFullCourtCase } from "types/display/CourtCases"
+import { CsrfTokenContext } from "context/CsrfTokenContext"
+
+interface Props {
+  courtCase: DisplayFullCourtCase
```
Comment (richrace) [inline src/components/ReallocationNotesForm.tsx:13]: You don't need this if you're using `CourtCaseConext`

File: src/pages/court-cases/[courtCaseId]/reallocate.tsx [hunk 3]
```diff
@@ -42,6 +46,8 @@ export const getServerSideProps = withMultipleServerSideProps(
       previousPath: string
     }
 
+    console.log("previousPath:", previousPath)
```
Comment (richrace) [inline src/pages/court-cases/[courtCaseId]/reallocate.tsx:49]: `console.log`

File: src/pages/court-cases/[courtCaseId]/reallocate.tsx [hunk 4]
```diff
@@ -87,96 +94,112 @@ export const getServerSideProps = withMultipleServerSideProps(
...
   const { basePath } = useRouter()
-  const currentForce = forces.find((force) => force.code === courtCase.orgForPoliceFilter?.substring(0, 2))
-  const forcesForReallocation = getForcesForReallocation(currentForce?.code)
-  const handleOnNoteChange: FormEventHandler<HTMLTextAreaElement> = (event) => {
-    setNoteRemainingLength(MAX_NOTE_LENGTH - event.currentTarget.value.length)
-  }
   const [currentUserContext] = useState<CurrentUserContextType>({ currentUser: user })
+  const [courtCaseContext] = useState<CourtCaseContextType>({ courtCase: courtCase })
+  const [csrfTokenContext] = useState<CsrfTokenContextType>({ csrfToken })
+  const [previousPathContext] = useState<PreviousPathContextType>({ previousPath })
```
Comment (richrace) [inline src/pages/court-cases/[courtCaseId]/reallocate.tsx:131]: Is this needed?

File: src/pages/court-cases/[courtCaseId]/reallocate.tsx [hunk 5]
```diff
@@ -87,96 +94,112 @@ export const getServerSideProps = withMultipleServerSideProps(
...
-                  </Button>
-                  <Link href={backLink}>{"Cancel"}</Link>
-                </ButtonsGroup>
-              </Fieldset>
-            </Form>
-          </ConditionalRender>
-        </Layout>
+        <CourtCaseContext.Provider value={courtCaseContext}>
+          <CsrfTokenContext.Provider value={csrfTokenContext}>
+            <PreviousPathContext.Provider value={previousPathContext}>
```
Comment (richrace) [inline src/pages/court-cases/[courtCaseId]/reallocate.tsx:159]: And here?

File: src/components/ReallocationNotesForm.tsx (hunk 2 above)
Comment (richrace) [reply src/components/ReallocationNotesForm.tsx:13]: See `useCourtCase` instead.

File: src/pages/court-cases/[courtCaseId]/reallocate.tsx (hunk 3 above)
Comment (C-gyorfi) [inline src/pages/court-cases/[courtCaseId]/reallocate.tsx:49]: forgot to remove this

**Review by C-gyorfi — COMMENTED** [review]
Any tests? 😄

File: src/pages/court-cases/[courtCaseId]/reallocate.tsx (hunk 3 above)
Comment (ian-antking) [reply src/pages/court-cases/[courtCaseId]/reallocate.tsx]: 😨

File: src/pages/court-cases/[courtCaseId]/reallocate.tsx (hunk 4 above)
Comment (ian-antking) [reply src/pages/court-cases/[courtCaseId]/reallocate.tsx]: I assumed so, but turns out no.

**Review by C-gyorfi — APPROVED** [review]

**Pull request merged**